### ISSUE TITLE
Repository Troubleshooting - invalid active developer path when running git

### ISSUE THREAD

**Issue by deleonl4** [susanBuck/e2-fall20#23]
I am troubleshooting creating my repository. It failed the first time. I am trying to accomplish step 2 list below but I can't find the e2 directory. I went to the document root directory (local server's htdocs directory): `/Applications/XAMPP/xamppfiles/htdocs` 

<img width="1056" alt="Screen Shot 2020-09-17 at 2 07 25 PM" src="https://user-images.githubusercontent.com/47702390/93510448-32b7cd00-f8ef-11ea-98bd-6f5a181b2861.png">

**Repository Troubleshooting Steps:**
1. Follow the steps under the Delete section in your repo’s settings on Github.
2. Remove the e2 directory on your computer from your document root folder.

This was the error I received in the terminal:

<img width="651" alt="Screen Shot 2020-09-17 at 2 12 50 PM" src="https://user-images.githubusercontent.com/47702390/93510920-f5077400-f8ef-11ea-973d-29f34f4ed041.png">

Two potential solutions but I just downloaded xcode... and when attempting 1 I received the error below.

1- `$ xcode-select --install`
2- `$ sudo xcode-select --reset`

<img width="531" alt="Screen Shot 2020-09-17 at 2 16 51 PM" src="https://user-images.githubusercontent.com/47702390/93511303-968ec580-f8f0-11ea-9bd9-bc667cbb7f82.png">

**Comment by susanBuck** [comment]
Hi @deleonl4 - Just an FYI I'm doing some research on this issue now. Will report back with more info shortly.

**Comment by deleonl4** [comment]
Thank you, I've been doing research too. I found this: https://stackoverflow.com/questions/19907576/xcode-is-not-currently-available-from-the-software-update-server which talks about updating xcode; I'm hesitant to delete the current xcode and update to a new one not knowing if that will render the correct results and how that might affect any work i've done thus far.

**Comment by susanBuck** [comment]
I'm finding lots of varying, sometimes conflicting, info regarding this issue. 

To get to the bottom of things, I want to gather some info...

Can you share the output of the following commands:

## Command 1
```
which git
```
This will tell us what git installation your system is trying to use.

## Command 2:
```
xcode-select -p
```
This will print the path of your active developer directory. Based on your screenshot, I expect this will be `/Library/Developer/CommandLineTools/usr/bin` but I want to double check. 

## Command 3:
```
ls /Library/Developer/CommandLineTools/usr/bin
```

This will output all the programs in your CommandLineTools (or tell us if no such directory exists).

Finally, try completely closing Terminal and re-opening it, then running `git` and see if you get the same error. I read one thing that said after you install xcode you have to restart Terminal for it to recognize it.

**Comment by deleonl4** [comment]
Same here in regard to the varying "solutions". 
Thank you, I shared all of the commands in the screenshot below.
I completely closed out of terminal and re-opened the app before running `git` but the results are the same.

<img width="583" alt="Screen Shot 2020-09-17 at 5 01 07 PM" src="https://user-images.githubusercontent.com/47702390/93528009-b4b3f000-f907-11ea-8641-2cf4426fbcdb.png">

<img width="585" alt="Screen Shot 2020-09-17 at 5 02 11 PM" src="https://user-images.githubusercontent.com/47702390/93528018-b7aee080-f907-11ea-992d-b861dab86f9f.png">

**Comment by susanBuck** [comment]
Based on the output you shared of `ls /Library/Developer/CommandLineTools/usr/bin` ("No such file or directory") it confirms you do not have the command line tools installed. 

To try and rectify that, goto <https://developer.apple.com/download/more/> and download the first option, *Command Line Tools of XCode 12*. Use the .dmg file it gives you from this download to install the command line tools.

Then, restart Terminal and try running `git` again.

If that doesn't work - would you like to set up a time to zoom/screenshare tomorrow (Friday) to debug this together? Let me know - I'm flexible most of the day.

**Comment by deleonl4** [comment]
There is no better feeling than resolving an issue. This did it. Thank you so much!! On to the next one, haha

**Comment by susanBuck** [comment]
Excellent! So happy to hear it.

Now that the pressing problem is solved, some background on why you had to go through this whole XCode/Command Line tools maze just to get git working...

There are different ways to install git.

+ One way is to download it directly (<https://git-scm.com/downloads>).
+ Another way is to get it via Apple's XCode, which comes with Git as part of its "command line developer tools"
+ Another way - if you don't want to the full install of XCode - is just to download XCode's "command line developer tools"

(There are additional ways beyond this, but the above are the only ones relevant to this discussion)

In the course notes, I state that if the command `git` is not found, you do not have it installed and you can download it directly (ref: https://hesweb.dev/e2/notes/git/setup#download-and-install-git).

This is a bit different from what you actually experienced, because when you ran git commands, you were told:

```
xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
```

Encountering this error, you smartly did some Googling for solutions, and found yourself on the path of sorting out Git via XCode/Command line developer tools. Because I think it's a good idea to have those command line developer tools installed in general, I went down that path with you (versus redirecting you to just downloading git directly).

All this is just to provide some background on why the notes/lecture doesn't cover this somewhat confusing process.
According to my research, this particular xcrun error is unique to Mac users who have upgraded to Catalina (ref: https://michaelsoolee.com/git-not-working-macos-catalina-xcrun/). I was trying to figure out why it wasn't something I faced when I updated to Catalina myself, and I think it's because I must be using an install of Git that wasn't dependent on XCode.

Note how my output of `which git` is:

```
/usr/local/git/bin/git
```

Whereas on your end it is:
```
/usr/bin/git
```

Long story short, because it wasn't something I ran into (and I hadn't had any other student report it in previous semesters), I wasn't aware it was a problem students might encounter when first working with Git - hence the lack of coverage. 

I'll be updating the notes to address this. :-)

**Comment by deleonl4** [comment]
Thank you! This one was a fun one to resolve. Your notes a super helpful and I know future students will really appreciate it.
